### PR TITLE
[venstarthermostat] Split away and schedule settings calls (resolves bug exiting away mode)

### PR DESCRIPTION
**Pull request by dae1804**
My ColorTouch thermostat has a newer firmware, which, from the OpenHab logs & experimenting with the REST API directly, I've observed to reject any POSTs to the settings endpoint that include the `schedule` parameter when the thermostat is currently in AWAY mode, regardless of the value passed for `schedule`.  This had the effect of preventing the OpenHab binding from un-setting away mode.

This is likely to be the same problem encountered by other users on this forum thread: https://community.openhab.org/t/venstar-binding-not-working/123780/4

This PR splits out the method used to perform thermostat settings updates, so that _only_ the value being changed is sent to the thermostat.  By avoiding the `schedule` parameter from being included at all when changing `away`, this ensures that the thermostat will not reject attempts to change away mode from away to home.

### CODE REVIEW

**Comment by digitaldan** [comment]
Thanks for the contribution!  I'll take a look at it over the weekend.

File: bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java [hunk 1]
```diff
@@ -382,13 +382,11 @@ private void setFanMode(VenstarFanMode fanmode) {
     }
 
     private void setAwayMode(VenstarAwayMode away) {
-        VenstarScheduleMode schedule = infoData.getScheduleMode();
-        updateSettings(away, schedule);
+        updateAwayMode(away);
```
Comment (digitaldan) [inline bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java:385]: Rather then call updateAwayMode, just put the logic from it in this function, no need to have 2

File: bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java [hunk 2]
```diff
@@ -382,13 +382,11 @@ private void setFanMode(VenstarFanMode fanmode) {
...
     }
 
     private void setScheduleMode(VenstarScheduleMode schedule) {
-        VenstarAwayMode away = infoData.getAwayMode();
-        updateSettings(away, schedule);
+        updateScheduleMode(schedule);
```
Comment (digitaldan) [inline bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java:389]: Rather then call updateScheduleMode, just put the logic from it in this function, no need to have 2

**Comment by digitaldan** [comment]
Hi @dae1804 thanks for the fix!  That was a tough one to track down, i appreciate you figuring it out and submitting a PR.  I had two comments about condensing the 4 functions into 2, but other then that i tested it out and it looks good so far.

**Comment by digitaldan** [comment]
Hi @dae1804 , any progress?

**Comment by dae1804** [comment]
Hi @digitaldan!  Apologies for the delay; I was out of town last week, and just got back late Sunday.  I'm still digging out of backlog, but should have this turned around this week, perhaps tomorrow.

**Comment by dae1804** [comment]
Hi @digitaldan!  Finally found a few minutes to get this turned around...  Methods merged, and I updated the comment as well.

**Review by digitaldan — APPROVED** [review]
LGTM, thanks!!!!

**Review by jlaur — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**